src/middlewares/product.middleware.js

```diff
@@ -0,0 +1,54 @@
+import { Products } from '../database/models/index';
+import { Categories } from '../database/models/index';
+import { uploadPhoto } from '../utils/cloudinary.util';
+
+export const checkIfProductExists = async (req, res, next) => {
+  const name = req.body.name;
+  const getProduct = await Products.findOne({
+    where: { name: name },
+  });
+  if (getProduct) res.status(404).json({ message: 'Product already exists' });
+  else next();
+};
+
+export const allowedExtensions = (...extensions) => {
+  return async (req, res, next) => {
+
+    const images = [];
+    for (let i = 0; i < req.files.images.length; i++) {
+      const ext = req.files.images[i].name.split('.').pop().toLowerCase();
+      if (extensions.includes(ext)) {
+        const { url } = await uploadPhoto(req, res, req.files.images[i]);
+        images.push(url);
+      }
+      else
+      return res.status(400).json({message:"Invalid extension"});
+    }
+    req.images = images;
+    next();
+  };
+};
+
+export const checkNumberOfImages = (req, res, next) => {
+  const images = req.images;
+   if (images.length <= 3)
+    return res.status(400).json({ message: 'Please insert atleat 4 images' });
+  next();
+};
+
+export const checkImageField = (req, res, next) => {
+  if (!req.files) {
+    res.status(400).json({message:'Please upload images'});
+  }else{
+    next();
+  }
+};
+
+export const checkCategory = async (req, res, next) => {
+  const category = req.body.category.toLowerCase();
+  const categories = await Categories.findOne({ where: { name: category } });
+  if (!categories)
+    return res.status(404).json({ message: "Product Category Doesn't Exist" });
+  req.body.categoryId = categories.id;
+  next();
+};
```

Comment:
- This is category middleware, not product middleware.
- Enforce destructuring assignment operator
- Your message should not be capitalize